README.md

```diff
@@ -50,59 +43,54 @@ public class SampleApplication {
-mesos.framework.name=sampleApp1
-```
-
-The purpose of `mesos.framework.name` is to distinguish instances of the scheduler.
+# Configuration
+Most of an application's boilerplate is provided my Mesos-Starter. To configure that boilerplate, there are a range of configuration options. This table describes those options, whether they are optional and whether they have a default value.
+
+Command                         | Description                                           | Default               | Required
+---                             | ---                                                   | ---                   | ---
+`spring.application.name`       | Application name                                      |                       | Yes
+`mesos.framework.name`          | Framework name used in Mesos and ZooKeeper            |                       | Yes
+`mesos.master`                  | Path to mesos master                                  |                       | Yes
+`mesos.zookeeper.server`        | IP:PORT of the zookeeper server                       |                       | Yes
+`mesos.resources.cpus`          | CPUs allocated to the task                            | Accept all            |
+`mesos.resources.mem`           | RAM allocated to the task                             | Accept all            |
+`mesos.resources.count`         | Number of task instances                              | As many as possible   |
+`mesos.resources.ports`         | A map of port mappings                                |                       |
+`mesos.resources.distinctSlave` | Reject offers from hosts that already have a task     | `false`               |
+`mesos.role`                    | Reject offers that are not offered to this role       | `*`                   |
+`mesos.docker.image`            | Docker image to use                                   |                       |
+`mesos.docker.network`          | Type of docker network                                | `BRIDGE`              |
+`mesos.command`                 | Set the docker CMD or shell command                   |                       |
+`mesos.uri`                     | A list of files to download into the Mesos sandbox    |                       |
+`mesos.principal`               | The Mesos principal for framework authentication      |                       |
+`mesos.secret`                  | The Mesos secret for framework authentication         |                       | 
+
+## Mesos master address format
+The `mesos.master` setting accepts any of the following formats: `host:port`, `zk://host1:port1,host2:port2,.../path`, `zk://username:password@host1:port1,host2:port2,.../path` or `file:///path/to/file`
+
+## Port Mapping
+The ports requirement serves two purposes. Requesting port resources from Mesos and mapping them to the application. The port settings must follow the format:
+```
+mesos.resources.ports.[NAME].{host,container}={ANY,UNPRIVILEGED,PRIVILEGED,[INTEGER]}
```

Comment:
Wouldn't it be clearer and easier to read if we split it into two lines?

```
mesos.resources.ports.[NAME].host={ANY,UNPRIVILEGED,PRIVILEGED,[INTEGER]}
mesos.resources.ports.[NAME].container=[INTEGER]
```

I think it makes it easier to understand that the `.container` is only required when `mesos.docker.image` != `null`.